feat: add C implementation for `number/float32/base/signbit`

Neerajpathak07 opened the pull request:
Resolves #3397 

## Description

> What is the purpose of this pull request?

This pull request:

-   add C implementations for `number/float32/base/signbit`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3397 & #649  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-  [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Neerajpathak07 committed:
feat: add benchmarks

Neerajpathak07 committed:
feat: add examples

Neerajpathak07 committed:
feat: add essential files

Neerajpathak07 committed:
feat: add tests

Neerajpathak07 committed:
fix: lint copyright years

Neerajpathak07 commented:
@Planeshifter  I just added all the essential files and building the logic but I am still facing issues withe the gyp native addons.
Would really appreciate a little help.

Neerajpathak07 committed:
fix: changing location

Neerajpathak07 committed:
fix: gyp errors

Neerajpathak07 committed:
fix: test errors

Planeshifter commented:
/stdlib check-required-files

Planeshifter commented:
@Neerajpathak07 

Check out the workflow failures. 

E.g., the lint step of `package.json` in https://github.com/stdlib-js/stdlib/actions/runs/12264198244/job/34217519484?pr=3399

```
ERROR: package.json in lib/node_modules/@stdlib/number/float32/base/signbit needs updates to directories and/or gypfile fields
diff --git a/lib/node_modules/@stdlib/number/float32/base/signbit/package.json b/lib/node_modules/@stdlib/number/float32/base/signbit/package.json
index 0b5a[8](https://github.com/stdlib-js/stdlib/actions/runs/12264198244/job/34217519484?pr=3399#step:10:9)f15d..1f40f6e35 100644
--- a/lib/node_modules/@stdlib/number/float32/base/signbit/package.json
+++ b/lib/node_modules/@stdlib/number/float32/base/signbit/package.json
@@ -14,11 +14,14 @@
     }
   ],
   "main": "./lib",
+  "gypfile": true,
   "directories": {
     "benchmark": "./benchmark",
     "doc": "./docs",
     "example": "./examples",
+    "include": "./include",
     "lib": "./lib",
+    "src": "./src",
     "test": "./test"
   },
   "types": "./docs/types",
```

tells us that several needed fields are not set and should be added.

Neerajpathak07 committed:
feat: update package.json

Neerajpathak07 commented:
@Planeshifter I have added the essential files in the dependency tree of package.json.
But I am having doubts on how to resolve this particular error:-
```
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make[1]: Entering directory '/home/runner/work/stdlib/stdlib/lib/node_modules/@stdlib/number/float32/base/signbit/build'
  CC(target) Release/obj.target/addon/src/addon.o
../src/addon.c: In function ‘addon’:
../src/addon.c:63:18: warning: implicit declaration of function ‘napi_get_value_float’; did you mean ‘napi_get_value_bool’? [-Wimplicit-function-declaration]
   63 |         status = napi_get_value_float( env, argv[ 0 ], &value );
      |                  ^~~~~~~~~~~~~~~~~~~~
      |                  napi_get_value_bool
  CC(target) Release/obj.target/addon/src/main.o
  CC(target) Release/obj.target/addon/../to-word/src/to_word.o
  SOLINK_MODULE(target) Release/obj.target/addon.node
  COPY Release/addon.node
  ACTION Copying addon... src/addon.node
  TOUCH Release/obj.target/copy_addon.stamp
make[1]: Leaving directory '/home/runner/work/stdlib/stdlib/lib/node_modules/@stdlib/number/float32/base/signbit/build'
gyp info ok 
```

Neerajpathak07 committed:
fix: src errors

Neerajpathak07 committed:
fix: lint errors

Neerajpathak07 committed:
fix: bench errors

Neerajpathak07 commented:
@Planeshifter All CI checks are passing pls review it. I am going to need this for implementing `atan2f`

Planeshifter commented:
@Neerajpathak07 Wil try taking a look and review tonight.

Planeshifter commented:
@Neerajpathak07 Wil try taking a look and review tonight.

Planeshifter commented on lib/node_modules/@stdlib/number/float32/base/signbit/manifest.json:
```diff
@@ -0,0 +1,40 @@
+{
+	"options": {},
```
JSON files should be indented with two spaces.

Planeshifter commented on lib/node_modules/@stdlib/number/float32/base/signbit/manifest.json:
```diff
@@ -0,0 +1,40 @@
+			"relative": false
+		}
+	],
+	"confs": [
```
Missing configs for examples and benchmarks in here.

Planeshifter reviewed (commented):
Thanks for your work on this! 

Looks pretty good to me. `manifest.json` will need some updates. @kgryte, you mind taking final look please and then merge the PR after the comments have been addressed?

Neerajpathak07 committed:
fix: manifest.json errors

Neerajpathak07 committed:
fix: CI errors

Neerajpathak07 committed:
fix: gyp errors

Neerajpathak07 committed:
fix: example errors

Neerajpathak07 commented:
@Planeshifter Thank you for the review.

Neerajpathak07 commented:
@kgryte any suggestions from your side??

Neerajpathak07 committed:
fix: updating manifest file

Neerajpathak07 commented:
@kgryte would really appreciate a review on this since a PR #3338 is on stand-by and awaiting this to be merged!

kgryte committed:
chore: clean-up

kgryte commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

kgryte commented:
/stdlib merge

kgryte reviewed (approved):
After a bit of clean-up, this PR now LGTM.

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into float32/signbit

kgryte committed:
style: remove trailing whitespace

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---
